src/main/java/org/mockito/internal/creation/bytebuddy/InlineByteBuddyMockMaker.java

```diff
@@ -271,6 +272,15 @@ public void resetMock(Object mock, MockHandler newHandler, MockCreationSettings
         }
     }
 
+    @Override
+    public void cleanUpMock(Object mock) {
```

Comment:
We are never calling this method with an actual mock. Is there a situation this should be the case? Could you maybe provide an example for it? (If it is not necessary, we might as well make this method `clearAllMocks` as well)

Reply:
I said that in comments before I updated the PR.

If we will go with clearAllMocks() we need an additional method if there is strong feature request for keeping some mocks alive later.

That might be over-engineering, but I'm not sure how open you are to introduce new method to incubating class or change existing incubating method signature in this case. It would require mock maker changes later.

Maybe we can change the public API to take in mocks as well? I might just do that quickly.

Reply:
Updated the PR, but I'm open to discuss this forward. If you two believe cutting off the mock specific clear up feature is better then I'm willing to make another change.

I'm just afraid that there are cases where callers might want to keep some mocks, but still need to release some other of them.

Reply:
I think it is fair to keep both. Since we are introducing an interface, we are most flexible if we keep them both in. That said, I think it is better to also introduce `clearAllMocks` on `InlineByteBuddyMockMaker` as well as rename `cleanUpMock` to `clearMock` to keep it consistent.

Reply:
Updated them for consistency. Thanks.